packages/seacas/libraries/ioss/src/Ioss_Region.C

```diff
@@ -2741,4 +2811,151 @@ namespace Ioss {
     }
   }
 
+  void Region::register_mesh_modification_observer(std::shared_ptr<DynamicTopologyObserver> observer)
+  {
+    if(observer && observer->get_control_option() == FileControlOption::CONTROL_AUTO_SINGLE_FILE) {
+      const Ioss::PropertyManager &properties = get_database()->get_property_manager();
+      if(!properties.exists("ENABLE_FILE_GROUPS")) {
+        std::ostringstream errmsg;
+        fmt::print(errmsg,
+                   "ERROR: File groups are not enabled in the database file '{}'.\n",
+                   get_database()->get_filename());
+        IOSS_ERROR(errmsg);
+      }
+    }
+
+    topologyObserver = observer;
+    topologyObserver->register_region(this);
+  }
+
+  void Region::start_new_output_database_entry(int steps)
+  {
+    if (get_database()->is_input())
+      return;
+
+    if(!topologyObserver)
+      return;
+
+    switch(topologyObserver->get_control_option()) {
+    case FileControlOption::CONTROL_AUTO_MULTI_FILE:
+      clone_and_replace_output_database(steps);
+      break;
+    case FileControlOption::CONTROL_AUTO_SINGLE_FILE:
+      add_output_database_group(steps);
+      break;
+    case FileControlOption::CONTROL_NONE:
+    default:
+      return;
+      break;
+    }
+  }
+
+  void Region::add_output_database_group(int steps, bool force_addition)
+  {
+    if (get_database()->is_input())
+      return;
+
+    if(!topologyObserver)
```

Comment:
The only true error here is calling this function when the output database has been opened without the groups  option. Currently, the option is obtained from the topologyObserver but it could also be inferred from the PropertyManager on the DatabaseIO object without the need of a registered topologyObserver.

I suppose I could add extra checks for these two scenarios to make sure that this is a valid call